### DISCUSSION TITLE
질문드려요

### DISCUSSION THREAD

**Discussion by blissJoseph** [yoonjeong-kwon/fastcampus-kubernetes#4] [Q&A] [answered]
안녕하세요.

Part3 쿠버네티스 입문 에서 Ch1. 03장 질문드려요.

쿠버네티스 클러스터 구성을 하는데 GCP 클러스터를 만들고 클라우드에서 제공하는 터미널까지 했습니다. 

그 다음으로 SDK로 설치를 하라고 설명을 하는데, 아래와 같은 질문을 드려요.

우선, 왜 SDK를 설치하는지 근원적으로 궁금하고 역할이 뭔지 알 수 있을까요?

클라우드는 AWS EC2랑 NCP 서버 정도만 사용해 봤습니다

**Comment by yoonjeong-kwon** [comment, marked as answer]
@blissJoseph 님 안녕하세요. 반갑습니다 :) 

Google Cloud SDK는 구글 클라우드에서 제공하는 서비스를 제어하기 위한 Command-line interface 입니다.
로컬 컴퓨터 등에서 이 명령줄 인터페이스를 이용해서 구글 클라우드 서버에 접속하거나 명령을 실행할 수 있습니다.

쿠버네티스 클러스터에 속한 서버는 구글 클라우드에서 제공하는 서버이고
**로컬 컴퓨터 등에서 구글 쿠버네티스 클러스터에 접속하기 위해서는 쿠버네티스 서버 주소, 서버 인증서 정보 등이 필요**합니다.
이 정보는 쿠버네티스 클러스터를 구성한 관리자, 즉 구글 클라우드에서 알려주어야 하는 값입니다. 
**구글 클라우드 서비스 사용자가 이 정보를 내려 받을 수 있도록 SDK를 통해 기능을 제공**하고 있습니다. 

슬라이드 27쪽에 보시면 `gcloud container clusters get-credentials <cluster이름> --zone <zone이름> --project <project이름>` 라는 명령줄 인터페이스 실행을 안내해드렸는데요.
`gcloud container clusters` 명령어 실행이 필요한 이유는 슬라이드 26, 28쪽에 쿠버네티스 설정 파일을 구성하기 위함입니다.
로컬 컴퓨터 -> 쿠버네티스 클러스터 접속 시 필요한 정보를 담은 kubeconfig 파일을 로컬 컴퓨터에 구성할 수 있도록 Google Cloud SDK를 통해 인터페이스를 제공하고 있습니다.

충분한 답변이 되셨을까요?
추가 질문 있으시면 또 남겨주세요.
감사합니다!
